src/Container/Container.php

```diff
@@ -0,0 +1,465 @@
+<?php
+
+/**
+ * This file is part of the Zest Framework.
+ *
+ * @author   Muhammad Umer Farooq <lablnet01@gmail.com>
+ * @author-profile https://www.facebook.com/Muhammadumerfarooq01/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ * @since 3.0.0
+ *
+ * @license MIT
+ */
+
+namespace Zest\Container;
+
+use Zest\Contracts\Container\Container as ContainerContract;
+
+class Container implements ContainerContract
+{
+	/**
+	 * Registered type hints.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $hints = [];
+
+	/**
+	 * Aliases.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $aliases = [];
+
+	/**
+	 * Singleton instances.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $instances = [];
+
+	/**
+	 * Contextual dependencies.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $contextualDependencies = [];
+
+	/**
+	 * Instance replacers.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $replacers = [];
+
+	/**
+	 * Parse the hint parameter.
+	 *
+	 * @param  string|array $hint Type hint or array contaning both type hint and alias
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return string
+	 */
+	protected function parseHint($hint): string
+	{
+		if(is_array($hint))
+		{
+			list($hint, $alias) = $hint;
+
+			$this->aliases[$alias] = $hint;
+		}
+
+		return $hint;
+	}
+
+	/**
+	 * Register a type hint.
+	 *
+	 * @param string|array    $hint      Type hint or array contaning both type hint and alias.
+	 * @param string|\Closure $class     Class name or closure.
+	 * @param bool            $singleton Should we return the same instance every time?
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function register($hint, $class, bool $singleton = false)
+	{
+		$this->hints[$this->parseHint($hint)] = ['class' => $class, 'singleton' => $singleton];
```

Comment:
Please check the `$hint` and `$class` type before assigning them to `$this->hints` array.

Reply:
Yes `$hint` will check in `parseHint` method
could you explain how we check the class parameter?

Reply:
I've mentioned some tips about type check on previous comment.

Reply:
so we can check like
```php
if (class_exists($class)) {
        //Do stuff....
}
```

Reply:
I think the type hints validation should be another method.

And it's responsible for validating type.

Reply:
You mean it validate is type hint is string or array?

Reply:
How about using the `instanceof`?

Reply:
Maybe the `class_exists` is fine, too.